dvc/dependency/param.py

```diff
@@ -88,12 +85,12 @@ def read_params(self):
         if not self.exists:
             return {}
 
+        suffix = self.path_info.suffix.lower()
+        parser = PARSERS[suffix]
         with self.repo.tree.open(self.path_info, "r") as fobj:
             try:
-                config = self.PARAMS_FILE_LOADERS[
-                    self.path_info.suffix.lower()
-                ](fobj)
-            except (yaml.YAMLError, toml.TomlDecodeError) as exc:
+                config = parser(fobj.read(), self.path_info)
+            except ParseError as exc:
```

Comment:
This is not perfect as it still needs a `tree` to open it correctly and could be error-prone (depending on how/what encoding it's opened, _et cetera_). So, in the next PR, I'm thinking of changing the serialization APIs to make it optionally aware of the trees.
```
def load_yaml(path, tree=None):
   pass
```

This way, I can make `parse_yaml` somewhat _internal_ and so that we could get away without using it for the most problems.

Reply:
Done in #4425